src/stream/src/task/barrier_manager.rs

```diff
@@ -431,14 +426,16 @@ impl LocalBarrierWorker {
                 }
                 completed_epoch = self.state.next_completed_epoch() => {
                     let result = self.on_epoch_completed(completed_epoch);
-                    self.control_stream_handle.inspect_result(result);
+                    if let Err(err) = result {
+                        self.notify_other_failure(err, "failed to complete epoch").await;
```

Comment:
Not that sure when this branch will be hit and whether it's also necessary to collect actor failures.